Sources/ImperialCore/FederatedServiceRouter.swift

```diff
@@ -111,33 +119,23 @@ extension FederatedServiceRouter {
 
         let buffer = try await body.encodeResponse(for: request).body.buffer
         let response = try await request.client.post(url, headers: self.callbackHeaders) { $0.body = buffer }
-        return try response.content.get(String.self, at: ["access_token"])
+        let token = try response.content.get(String.self, at: ["access_token"])
+        return (token, response.body)
+    }
+    
+    package func verifyState(_ request: Request) throws {
+        guard let requestState = request.query[String.self, at: "state"],
+              let sessionState = request.session.state,
+              requestState == sessionState else {
+            throw Abort(.badRequest)
+        }
     }
 
     package func callback(_ request: Request) async throws -> Response {
-        let accessToken = try await self.fetchToken(from: request)
+        let (accessToken, responseBody) = try await self.fetchTokenAndResponseBody(from: request)
         let session = request.session
         try session.setAccessToken(accessToken)
-        let response = try await self.callbackCompletion(request, accessToken)
+        let response = try await self.callbackCompletion(request, accessToken, responseBody)
         return try await response.encodeResponse(for: request)
-    }
-}
-
-/// Convenience URLQueryItems
-extension FederatedServiceRouter {
```

Comment:
I removed these and created an extension on URLQueryItem to:
.init(clientID:)
.init(redirectURIItem:)
static var codeResponseType: URLQueryItem